File: binsV1/binV24/second_problem.js
const posts = {
  post_1: { tags: ["js", ["cli", "json"], "parsing"] },
  post_2: { tags: ["js", ["api", ["cli"]]] },
};

const tags = Object.values(posts).flatMap((post) => post.tags.flat(5));

const uniqueTags = [...new Set(tags)];

console.log(JSON.stringify({ uniqueTags }, null, 2));

File: binV3/second_problem.rb
require 'json'

input_data = {
  "post_1": { "tags": [ "ruby", [ "cli", "json" ], "parsing" ] },
  "post_2": { "tags": [ "ruby", [ "api", [ "cli" ] ] ] }
}

tags = input_data.values.flat_map do |data|
  data[:tags].flatten
end.uniq

print JSON.pretty_generate(tags)
